src/main/java/com/crowdin/client/Crowdin.java

```diff
@@ -0,0 +1,210 @@
+            Credentials credentials = new Credentials(crowdinClientProperties.getToken(), null, crowdinClientProperties.getBaseUrl());
+            ClientConfig clientConfig = ClientConfig.builder()
+                    .userAgent("crowdin-android-studio-plugin/ " + PluginManager.getPlugin(PluginId.getId(PluginManager.CORE_PLUGIN_ID)).getVersion() + " android-studio/" + PluginManager.getPlugin(PluginId.getId("com.crowdin.crowdin-idea")).getVersion())
+                    .httpClient(new ApacheHttpClient(credentials, new JacksonJsonTransformer(), Collections.emptyMap()))
+                    .build();
+            this.client = new Client(credentials, clientConfig);
+        }
+    }
+
+    public void uploadFile(VirtualFile source, String branch) {
+        if (source == null || this.invalidConfiguration) {
+            return;
+        }
+
+        try {
+            Long branchId = this.getOrCreateBranch(branch);
+
+            ResponseObject<Storage> storageResponseObject = this.client.getStorageApi().addStorage(source.getName(), source.getInputStream());
+            Long storageId = storageResponseObject.getData().getId();
+            ResponseList<com.crowdin.client.sourcefiles.model.File> fileResponseList = this.client.getSourceFilesApi().listFiles(this.projectId, branchId, null, null, 500, null);
+            ResponseObject<com.crowdin.client.sourcefiles.model.File> foundFile = fileResponseList.getData().stream()
+                    .filter(f -> f.getData().getName().equals(source.getName()))
+                    .findFirst().orElse(null);
+            if (foundFile != null) {
+                UpdateFileRequest request = new UpdateFileRequest();
+                request.setStorageId(storageId);
+                GeneralFileExportOptions generalFileExportOptions = new GeneralFileExportOptions();
+                generalFileExportOptions.setExportPattern("/values-%android_code%/%original_file_name%");
+                this.client.getSourceFilesApi().updateOrRestoreFile(this.projectId, foundFile.getData().getId(), request);
+            } else {
+                AddFileRequest request = new AddFileRequest();
+                request.setStorageId(storageId);
+                request.setName(source.getName());
+                GeneralFileExportOptions generalFileExportOptions = new GeneralFileExportOptions();
+                generalFileExportOptions.setExportPattern("/values-%android_code%/%original_file_name%");
+                request.setExportOptions(generalFileExportOptions);
+                this.client.getSourceFilesApi().addFile(this.projectId, request);
+            }
+        } catch (Exception e) {
+            NotificationUtil.showErrorMessage(this.getErrorMessage(e));
+        }
+    }
+
+    public File downloadTranslations(VirtualFile sourceFile, String branch) {
+        if (this.invalidConfiguration) {
+            return null;
+        }
+        try {
+            Long branchId = this.getOrCreateBranch(branch);
```

Comment:
Perhaps this is more to the logic of command and not to API update, but there is no need to create a branch and download a project if this branch doesn't exist.

Reply:
fixed